compositor_common/src/frame.rs

```diff
@@ -1,15 +1,48 @@
-use std::time::Duration;
+use std::{collections::HashMap, sync::Arc, time::Duration};
 
 use crate::scene::Resolution;
 
-#[allow(dead_code)]
+// Clone is temporary. Testing use only.
 #[derive(Debug, Clone)]
 pub struct Frame {
     pub data: YuvData,
     pub resolution: Resolution,
     pub pts: Duration,
 }
 
+pub type Pts = Duration;
+pub type InputID = u32;
+
+/// TODO: This should be a rational.
+#[derive(Debug, Clone, Copy)]
+pub struct Framerate(pub u32);
```

Comment:
I suppose, that you wanted to keep the duration between two frames as framerate (e.g. `1/30s` with `30fps`).

Theoretically, this causes rounding error accumulation in the `pts` calculation of the next `"input frame batches"` (With some framerates, the exact interval between two frames has no precise representation with natural numbers, e.g. `30fps`, `60fps`). Using framerate directly, we can avoid this error accumulation with:
```rust
pub struct InternalQueue {
    ...
    output_framerate: Framerate,
    pub send_batches_counter: u32,
}

impl InternalQueue {
...

    pub fn get_next_output_buffer_pts(&self) -> Duration {
        Duration::from_secs_f64(self.send_batches_counter as f64 / self.output_framerate.0 as f64)
    }
}
```

In reality, it shouldn't be a problem, since with each frame we would accumulate at most one nanosecond of error, which would be hardly noticeable even after a few days of constant streaming.

Aside from that, I think it's simpler to keep framerate as `u32` or some `rational`, than as `Duration`. With `Duration` it isn't obvious to me what it should represent. I only guessed that you meant the interval between the next two frames. 
I wouldn't use `Duration` here mostly for clarity.